### ISSUE TITLE
(Possibly) redundant usage of 7z

### ISSUE THREAD

**Issue by lipowskm** [emkor/napi-py#1]
So one thing I noticed is that you could get rid of the necessity of having `7z` as an external tool completely.

You can manage it using `py7zlib` package. Here's a sample code how I'm using it:

```python
from io import BytesIO
import base64
import xml.etree.ElementTree as ET
from py7zlib import Archive7z

response = requests.post(
    url='http://www.napiprojekt.pl/api/api-napiprojekt3.php',
    data=dict(
        downloaded_subtitles_id=movie_hash,
        downloaded_subtitles_lang='PL',
        client='NapiProjekt',
        mode=17,
    )
)
parser = ET.XML(response.content)
content = parser.find('subtitles').find('content').text
decoded_content = base64.b64decode(content)
buffer = BytesIO(decoded_content)
archive = Archive7z(buffer, password='iBlm8NTigvru0Jr0')
encoded_subtitles = archive.getmember(0).read()
subtitles = encoded_subtitles.decode('Windows-1250 or some other encoding')
```

Don't ask me about the `password` since I don't know where it came from, I found this piece of code somewhere on Github and did a little refactor, since it was outdated. But it does it's job obviously and doesn't require `7z` at all.

**Comment by lipowskm** [comment]
Ok now I've noticed that you're using that password too, my bad.
Have you considered switching to the package instead of using 7z?

**Comment by emkor** [comment]
Honestly, I have not, but maybe it's a good idea. Two things I worry about:
- availability of the package on other platforms (arm/x86, macOS/linux)
- number and size of dependencies of the package (would be nice if it require only builtins)

Subtitles are small files anyway so decompression performance is not that important, I think.

I can work on this but not before weekend, if you @lipowskm have the time and dedication to implement it sooner, you're welcome to open first PR

**Comment by lipowskm** [comment]
The package itself is called `pylzma` and it doesn't require any dependencies, so it's a good thing.
Also from the package homepage:

![obraz](https://user-images.githubusercontent.com/54337215/100870175-778ae300-349e-11eb-95e4-4478843a09ad.png)

It states that it's platform independent. Hopefully I will have time to implement it before the weekend.

**Comment by lipowskm** [comment]
PR is now opened.

**Comment by emkor** [comment]
Thanks for implementation, updated version is now published on pypi as `1.1.2` after merging https://github.com/emkor/napi-py/pull/2
